pyVoIP/SIP/client.py

```diff
@@ -417,6 +417,8 @@ def gen_digest(
         user = request.headers["From"]["user"]
         credentials = self.credentials_manager.get(server, realm, user)
         username = credentials["username"]
+        if request.authentication["header"].lower() == "proxy-authenticate":
+            username = credentials["user"]
```

Comment:
Shouldn't this be `credentials["username"]`?

Reply:
The main issue is that `CredentialsManager.get` doesn't return `user`, only `username` and `password`.

```py
from pyVoIP.credentials import CredentialsManager

cm = CredentialsManager()
cm.add("username", "password", server="google.com", realm="realm", user="user")
assert cm.get("google.com", "realm", "user") == {
    "username": "username",
    "password": "password",
}
```
If you are calling from `<sip:100@mycomputer.com>` and the result should be `Proxy-Authentication: Digest username="ujjkhe986",realm="...",nonce="...",uri="...",response="...",algorithm="..."` then the proper configuration would be:

```py
from pyVoIP.credentials import CredentialsManager

cm = CredentialsManager()
cm.add("ujjkhe986", "mysUperS3crEtP4s5w02D", user="100")
```

So unless you also modified `credentials = self.credentials_manager.get(server, realm, user)` and did not commit it here, `username = credentials["user"]` will raise `KeyError: 'user'`

Reply:
thx for clarifying the credentials manager, switched back to draft and in this case the mentioned lines are indeed obsolete